pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -100,6 +102,30 @@ def __verify_eval_loop_configuration(self, model: 'pl.LightningModule', stage: s
             rank_zero_warn(f'you passed in a {loader_name} but have no {step_name}. Skipping {stage} loop')
         if has_step and not has_loader:
             rank_zero_warn(f'you defined a {step_name} but have no {loader_name}. Skipping {stage} loop')
+        if has_step and has_loader:
+            sig_args = inspect.signature(getattr(model, step_name)).parameters
+            len_args = len(sig_args)
+            loaders = getattr(model, loader_name)()
+            len_loaders = 1 if not isinstance(loaders, list) else len(loaders)
+            if 'args' in sig_args:  # positional arg work with both single and multi dataloader
+                pass
+            if 'args' not in sig_args and 'kwargs' in sig_args:
+                raise MisconfigurationException(
+                    f'Method {step_name} does not work with single `kwargs` argument. Either change the signature'
+                    f' to `{step_name}(self, batch, batch_idx, dataloader_idx=0)`'
+                    f' or `{step_name}(self, *args, **kwargs)`'
```

Comment:
Why is this a limitation?

I would expect  `self, batch, batch_idx, **kwargs` to work by passing the dataloader_idx argument.